OneCommunityGlobal/HighestGoodNetworkApp
# Yongjian fix new max personal record award badge

# Description
![Screen Shot 2023-05-13 at 5 21 34 PM](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/46613683/aca457bc-0fff-4c19-8822-cf6b2e2133af)


## Mainly changes explained:
Added a conditional check that if the sum of reported tangible hours + volunteer time is greater than the previous personal best max hours, we update the personal best max hours and assign the user a "New Max - Personal Record" badge with no rounding up. The issue of duplicate badges have also been fixed (e.g. instead of the same badge displaying 20 times, now the number "20" appearing within a badge if users have earned the same badge 20 times).

## How to test:
1. check into current branch
2. do `npm run start:local` to run this PR locally
3. check into backend's development branch and do `npm run build` and `npm start`
4. login as admin role and start timer and stop the timer, the New Max - Personal Record" badge should display the your personal maximum hours volunteered in one week. Note: the "New Max - Personal Record" badge count under "New Badges Earned".

Note: if you don't see the changes immediately, please restart the servers and logout and login again. Another way is to clear the cache.

## Screenshots
![Screen Shot 2023-05-17 at 7 32 45 PM](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/46613683/f1d0b032-730b-4d47-96ce-0df2fe06bb8a)
![Screen Shot 2023-05-17 at 7 34 06 PM](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/46613683/c3f4281b-3a8e-4b2b-9cf1-f18226223bda)
![Screen Shot 2023-05-13 at 1 33 04 PM](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/46613683/f694da39-b7d4-4900-bbf6-71bb0a47d7b9)
![Screen Shot 2023-05-13 at 5 00 27 PM](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/46613683/53d94731-0ecd-4413-8319-02d3a5e19145)
